### PR TITLE
Add gocritic to linters, fix linter configuration

### PR DESCRIPTION
**Pull request by mem**
scripts/go/configs/golangci.yml has had a typo since forever and the
configuration contained in there wasn't taking effect.

After fixing the configuration a bunch of diagnostics showed up.

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

### CODE REVIEW

File: cmd/synthetic-monitoring-agent/main.go [hunk 1]
```diff
@@ -152,7 +151,7 @@ func run(args []string, stdout io.Writer) error {
 		Features:       features,
 	})
 	if err != nil {
-		log.Fatalf("Cannot create checks updater: %s", err)
+		return fmt.Errorf("Cannot create checks updater: %w", err)
```
Comment (mem) [inline cmd/synthetic-monitoring-agent/main.go:154]: this is the linter pointing out that defer blocks won't run

File: internal/pkg/prom/prom.go [hunk 2]
```diff
@@ -287,11 +287,12 @@ func NewTLSConfig(cfg *TLSConfig) (*tls.Config, error) {
 		tlsConfig.ServerName = cfg.ServerName
 	}
 	// If a client cert & key is provided then configure TLS config accordingly.
-	if len(cfg.CertFile) > 0 && len(cfg.KeyFile) == 0 {
+	switch {
```
Comment (mem) [inline internal/pkg/prom/prom.go:290]: this is the linter saying that this is easier to read as a switch statement

File: pkg/pb/synthetic_monitoring/checks_extra.go [hunk 3]
```diff
@@ -167,68 +169,12 @@ func (c *Check) Validate() error {
...
-		settingsCount++
-	}
-
-	if c.Settings.Traceroute != nil {
-		settingsCount++
-	}
-
-	if settingsCount != 1 {
-		return ErrInvalidCheckSettings
+	if err := c.validateTarget(); err != nil {
```
Comment (mem) [inline pkg/pb/synthetic_monitoring/checks_extra.go:172]: this is the linter saying that this function is too complex

File: pkg/pb/synthetic_monitoring/checks_extra_test.go [hunk 4]
```diff
@@ -215,6 +215,19 @@ func TestCheckType(t *testing.T) {
 			},
 			expected: CheckTypeTcp,
 		},
+		"traceroute": {
```
Comment (mem) [inline pkg/pb/synthetic_monitoring/checks_extra_test.go:218]: this was missing

File: pkg/pb/synthetic_monitoring/checks_extra_test.go [hunk 5]
```diff
@@ -254,6 +267,10 @@ func TestCheckTypeString(t *testing.T) {
 			input:    CheckTypeTcp,
 			expected: "tcp",
 		},
+		"traceroute": {
```
Comment (mem) [inline pkg/pb/synthetic_monitoring/checks_extra_test.go:270]: this was missing

File: scripts/go/configs/golangci.yml [hunk 6]
```diff
@@ -37,19 +37,59 @@ linters:
     - varcheck
     - whitespace
 
-linter-settings:
+linters-settings:
```
Comment (mem) [inline scripts/go/configs/golangci.yml:40]: this either change in some version of golangci-lint, or it never worked in the first place

File: scripts/go/configs/golangci.yml [hunk 7]
```diff
@@ -37,19 +37,59 @@ linters:
...
   goconst:
-    ignore-tests: true
+    ignore-tests: false
     min-len: 5
     min-occurrences: 5
   gocyclo:
-    min-complexity: 10
+    min-complexity: 20
```
Comment (mem) [inline scripts/go/configs/golangci.yml:46]: we can progressively lower this down to 10

File: scripts/go/configs/golangci.yml [hunk 8]
```diff
@@ -37,19 +37,59 @@ linters:
...
   golint:
-    ignore-tests: true
+    ignore-tests: false
     min-confidence: 3
   rowserrcheck:
     packages:
       - github.com/jmoiron/sqlx
+  gocritic:
+    enabled-checks:
```
Comment (mem) [inline scripts/go/configs/golangci.yml:54]: this long list is the default checks + ruleguard

File: scripts/go/dep.go [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package main
+
+import (
+	_ "github.com/golangci/golangci-lint/cmd/golangci-lint"
```
Comment (mem) [inline scripts/go/dep.go:4]: this is trying to force go.mod to keep a sane list of dependencies

File: scripts/go/configs/golangci.yml (hunk 7 above)
Comment (electron0zero) [reply scripts/go/configs/golangci.yml:46]: can we add a TODO for this please :)

**Review by electron0zero — APPROVED** [review]

**Pull request merged**